brut.j.dir/src/main/java/brut/directory/ZipRODirectory.java

```diff
@@ -120,7 +120,7 @@ private void loadAll() {
             ZipEntry entry = entries.nextElement();
             String name = entry.getName();
             
-            if (name.equals(getPath()) || ! name.startsWith(getPath())) {
+            if (name.equals(getPath()) || ! name.startsWith(getPath()) || name.contains(".." + separator)) {
```

Comment:
Line 139 seems to be guaranteed to be false now. You might want to check the logic down there.

Reply:
Good catch. Commit for myself to look back at - https://github.com/iBotPeaches/Apktool/commit/693f592b137b8289fc80a7c14baf4c409564c0c8